addon/components/frost-text.js

```diff
@@ -1,51 +1,105 @@
-import FrostEvents from '../mixins/frost-events'
 import computed from 'ember-computed-decorators'
+import Ember from 'ember'
+const {
+  Component,
+  isPresent,
+  on
+} = Ember
+import {
+  task,
+  timeout
+} from 'ember-concurrency'
+import FrostEventsProxy from '../mixins/frost-events-proxy'
+import layout from '../templates/components/frost-text'
 
-const {TextField} = Ember
-const {next, schedule} = Ember.run
-
-const svgNs = 'http://www.w3.org/2000/svg'
-const clearSvg = `
-  <svg class='frost-text-clear' fill="#000000" height="24" viewBox="0 0 24 24" width="24" xmlns="${svgNs}">
-    <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/>
-    <path d="M0 0h24v24H0z" fill="none"/>
-  </svg>
-`
+export default Component.extend(FrostEventsProxy, {
 
-export default TextField.extend(FrostEvents, {
   // == Properties =============================================================
-  attributeBindings: [
-    'style'
-  ],
+
+  align: 'left', // TODO PropTypes
   classNames: [
     'frost-text'
   ],
+  classNameBindings: [
+    'isClearVisible',
+    'isClearEnabled'
+  ],
+  eventVersion: null, // TODO Deprecation message
+  isClearEnabled: false, // TODO PropTypes
+  isClearVisible: false, // TODO PropTypes
+  layout,
+  tabindex: 0, // TODO PropTypes
+  type: 'text', // TODO PropTypes
+
+  // == Computed properties  ===================================================
 
-  // == Computed properties ====================================================
-  @computed('align')
-  style (align) {
-    return Ember.String.htmlSafe(`text-align: ${CSS.escape(align)}`)
+  @computed('value', '_focused')
+  isFocusedWithValue (value, _focused) {
+    return this._focused && isPresent(this.value)
   },
 
-  // == Event hooks ============================================================
+  // == Events =================================================================
 
-  didInsertElement () {
-    schedule('render', this, function () {
-      this.$().after(clearSvg)
-      next(this, function () {
-        this.$().next().click(() => {
-          this.onClear()
-        })
+  _focusIn: on('focusIn', function () {
+    this.set('_focused', true)
+    this.get('_toggleClear').perform(this.get('isFocusedWithValue'))
+  }),
+
+  _focusOut: on('focusOut', function (event) {
+    this.set('_focused', false)
+    this.get('_toggleClear').perform(this.get('isFocusedWithValue'))
+  }),
+
+  _input: on('input', function (event) {
+    this.get('_toggleClear').perform(this.get('isFocusedWithValue'))
+  }),
+
+  // == Tasks ==================================================================
+
+  _toggleClear: task(function * (isVisible) {
+    if (isVisible) {
+      this.setProperties({
+        isClearEnabled: true,
+        isClearVisible: true
       })
-    })
-  },
+    } else {
+      this.set('isClearVisible', false)
+      yield timeout(200)
+      this.set('isClearEnabled', false)
```

Comment:
I'll add comments - focus events are a pain.  Basically we get the focusOut on the input prior to the click on the clear, which gets processed and hides the clear, so it's not available to click on.  This prolongs the clickability of the clear until it's fully transparent.